http-cors/src/main/scala/org/apache/pekko/http/cors/javadsl/CorsDirectives.scala

```diff
@@ -0,0 +1,36 @@
+package org.apache.pekko.http.cors.javadsl
+
+import java.util.function.Supplier
+
+import org.apache.pekko
+import pekko.http.javadsl.server.{ RejectionHandler, Route }
+import pekko.http.javadsl.server.directives.RouteAdapter
+import pekko.http.cors.javadsl.settings.CorsSettings
+import pekko.http.cors.scaladsl
+
+object CorsDirectives {
+
+  def cors(inner: Supplier[Route]): Route =
+    RouteAdapter {
+      scaladsl.CorsDirectives.cors() {
+        inner.get() match {
+          case ra: RouteAdapter => ra.delegate
+        }
+      }
+    }
+
+  def cors(settings: CorsSettings, inner: Supplier[Route]): Route =
+    RouteAdapter {
+      // Currently the easiest way to go from Java models to their Scala equivalent is to cast.
+      // See https://github.com/akka/akka-http/issues/661 for a potential opening of the JavaMapping API.
+      val scalaSettings = settings.asInstanceOf[scaladsl.settings.CorsSettings]
```

Comment:
Now that the code lives inside pekko, we could actually use `JavaMapping`, if this comment is still relevant. At least the comment should be updated to not point to akka?

Reply:
Fixed in latest push